Front-end

BourgoisMickael opened the pull request.

BourgoisMickael committed:
[Front] Fixed eslint error

BourgoisMickael committed:
[Server] Added a cors middleware

BourgoisMickael committed:
[Front] Added store and implemented manifest in it

BourgoisMickael committed:
[Front] Implemented build's store

Flora-Ht committed:
[Front] Separated builds and manifests

Flora-Ht committed:
[Front] Added basic create manifest vue

Flora-Ht committed:
[Front] Added basic manifest list and details

Flora-Ht committed:
[Front] Added select for manifests in create build

Flora-Ht committed:
[Front] Changed presentation of editor for creating manifest

Flora-Ht committed:
[Front] Changed pres

BourgoisMickael committed:
[Front] Retrieved and added front folder from previous versions

BourgoisMickael committed:
[Front] Fixed lint error and implemented store in home builds

BourgoisMickael committed:
[Front] Implemented store in details build

BourgoisMickael committed:
[Front] Implemented store in create build

BourgoisMickael committed:
[Front] Implemented store in home manifest

BourgoisMickael committed:
[Front] Implemented store on detail manifest

BourgoisMickael committed:
[Front] Implemented store on create manifest

BourgoisMickael committed:
[Front] Added output in build details

BourgoisMickael committed:
[Front] Fixed lint error

BourgoisMickael committed:
[Builder] Changed nest_server_api_url not pointing to the api

BourgoisMickael committed:
[Front] Set manifest content in editor at upload

BourgoisMickael committed:
[Front] Fixed table in detail build

BourgoisMickael committed:
[Front] Fixed table on detail manifest

BourgoisMickael committed:
[Front] Fixed build state in build listing

BourgoisMickael committed:
[Front] Moved views in subfolders

BourgoisMickael committed:
[Front] Refactored Views by using store

Improve views for build (home/create/details)
Improve views for manifest (home/create/details)

BourgoisMickael committed:
[Front] Added multiple history in manifest details view

BourgoisMickael committed:
[Front] Added update manifest page

BourgoisMickael committed:
[Front] Added pagination on manifest/build list

Arignir commented on front/README.md:
```diff
@@ -0,0 +1,28 @@
+## Getting Started
```
The content of that file should probably be moved to the root `README.md`

BourgoisMickael replied:
There will be a readme for each part of build (front/server/builder) and the front readme will have a link to each project otherwise the root readme wiil be too long

Arignir replied:
Do it now please, otherwise it's a bit of a mess :/

BourgoisMickael replied:
I made a root README with links to other readme

Arignir commented on front/public/index.html:
```diff
@@ -0,0 +1,25 @@
+  <div id="app"></div>
+  
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.22.2/moment.min.js"></script>
+  
```
That blank line should be removed

Arignir commented on front/src/app.vue:
```diff
@@ -0,0 +1,125 @@
+          </b-navbar-brand>
+          <b-collapse id="nav_collapse" is-nav>
+            <b-navbar-nav class="ml-auto">
+              <b-nav-item to="/" exact class="nav-item" active-class="nav-item-active">Builds</b-nav-item>
```
I'm not a fan of having those in the navbar, but I guess we'll fix it after this PR with @Queeniie

BourgoisMickael replied:
marked as todo

Arignir commented on front/src/router.js:
```diff
@@ -0,0 +1,26 @@
+  mode: 'history',
+  routes: [
+    { path: '/', component: HomeBuilds },
+    { path: '/builds/create', component: CreateBuild },
```
Same, we can improve those URLs but we'll do that later

BourgoisMickael replied:
marked as todo

Arignir commented on front/src/views/build/create.vue:
```diff
@@ -0,0 +1,228 @@
+        <h1>Create build</h1>
+        <!-- Error handling -->
+        <div v-if="getBuildLoadings.create" class="loading">
+          adding the build...
```
```suggestion
          Adding the build...
```

Arignir commented on front/src/views/build/create.vue:
```diff
@@ -0,0 +1,228 @@
+          adding the build...
+        </div>
+        <div v-if="error || getBuildErrors.create" class="build-error">
+          <p>An error has occured during the creation of the build</p>
```
```suggestion
          <p>An error occurred preventing the creation of the build</p>
```

Arignir commented on front/src/views/build/create.vue:
```diff
@@ -0,0 +1,228 @@
+        </div>
+
+        <div v-if="getManifestLoadings.list" class="loading">
+          loading
```
```suggestion
          Loading
```

Arignir commented on front/src/views/build/create.vue:
```diff
@@ -0,0 +1,228 @@
+          loading
+        </div>
+        <div v-else-if="getManifestErrors.list" class="build-error">
+          <p>An error has occured, please refresh or try again later</p>
```
```suggestion
          <p>An error occurred, please try again later</p>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+
+      <!-- error handling -->
+      <div v-if="getBuildLoadings.get" class="loading">
+        retrieving build...
```
```suggestion
        Retrieving the build's data...
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+        retrieving build...
+      </div>
+      <div v-if="getBuildErrors.get" class="build-error">
+        <p>An error has occured during the retrieving of build #{{ id }}</p>
```
```suggestion
        <p>An error occurred during while retrieving build #{{ id }}</p>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+          <tr>
+            <th>Id</th>
+            <th>State</th>
+            <th>exit_status</th>
```
```suggestion
            <th>Exit Status</th>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+            <th>Id</th>
+            <th>State</th>
+            <th>exit_status</th>
+            <th>creation date</th>
```
```suggestion
            <th>Creation Date</th>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+            <th>State</th>
+            <th>exit_status</th>
+            <th>creation date</th>
+            <th>start date</th>
```
```suggestion
            <th>Starting Date</th>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+            <th>exit_status</th>
+            <th>creation date</th>
+            <th>start date</th>
+            <th>end at</th>
```
Would be better to indicate the duration instead of the finish date, but same we can fix that later

BourgoisMickael replied:
marked as todo

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+            <th>exit_status</th>
+            <th>creation date</th>
+            <th>start date</th>
+            <th>end at</th>
```
```suggestion
            <th>Finish date</th>
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+          tdClass: 'list-table-cell',
+          formatter: (value) => {
+            switch (value) {
+              case 0:
```
Please avoid magic values and name them somewhere

BourgoisMickael replied:
Removed dateFields

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+              case 1:
+                return 'Building'
+              case 2:
+                return 'Failed'
```
```suggestion
                return 'Fail'
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+              case 2:
+                return 'Failed'
+              case 3:
+                return 'Successful'
```
```suggestion
                return 'Success'
```

Arignir commented on front/src/views/build/details.vue:
```diff
@@ -0,0 +1,233 @@
+      return this.getBuild(this._id)
+    },
+    packages () {
+      return (this.build && this.build.packages && this.build.packages.length && this.build.packages) || []
```
```suggestion
      return (this.build && this.build.packages && this.build.packages.length) || []
```

Arignir commented on front/src/views/build/home.vue:
```diff
@@ -0,0 +1,387 @@
+
+    <!-- error handling -->
+    <div v-if="getBuildLoadings.list" class="loading">
+      listing builds...
```
```suggestion
      Looking for builds...
```

Arignir commented on front/src/views/build/home.vue:
```diff
@@ -0,0 +1,387 @@
+      listing builds...
+    </div>
+    <div v-if="getBuildErrors.list" class="build-error">
+      <p>An error has occured during the retrieving of builds</p>
```
```suggestion
      <p>An error occurred while retrieving the list of builds</p>
```

Arignir commented on front/src/views/build/home.vue:
```diff
@@ -0,0 +1,387 @@
+              <b-row>
+                <b-col>
+                  <div class="item-info">
+                    <div v-if="item.state === 3">
```
Same as before, please avoid magic values and name them using an enumeration

BourgoisMickael replied:
There is now an enumeration for the state

Arignir commented on front/src/views/manifest/create.vue:
```diff
@@ -0,0 +1,281 @@
+        <h1>Create manifest</h1>
+        <!-- Error handling -->
+        <div v-if="getManifestLoadings.create" class="loading">
+          adding the manifest...
```
```suggestion
         Creating the manifest...
```

Arignir commented on front/src/views/manifest/create.vue:
```diff
@@ -0,0 +1,281 @@
+          adding the manifest...
+        </div>
+        <div v-if="error || getManifestErrors.create" class="build-error">
+          <p>An error has occured during the creation of the manifest</p>
```
```suggestion
          <p>An error occurred while creating the manifest</p>
```

Arignir commented on front/src/views/manifest/details.vue:
```diff
@@ -0,0 +1,189 @@
+
+      <!-- error handling -->
+      <div v-if="getManifestLoadings.get" class="loading">
+        retrieving manifest...
```
```suggestion
        Retrieving manifest...
```

Arignir commented on front/src/views/manifest/details.vue:
```diff
@@ -0,0 +1,189 @@
+        retrieving manifest...
+      </div>
+      <div v-if="getManifestErrors.get" class="build-error">
+        <p>An error has occured during the retrieving of manifest #{{ id }}</p>
```
```suggestion
        <p>An error occurred while retrieving manifest #{{ id }}</p>
```

Arignir commented on front/src/views/manifest/home.vue:
```diff
@@ -0,0 +1,313 @@
+
+    <!-- error handling -->
+    <div v-if="getManifestLoadings.list" class="loading">
+      listing manifests...
```
```suggestion
      Retrieving the list of manifests...
```

Arignir commented on front/src/views/manifest/home.vue:
```diff
@@ -0,0 +1,313 @@
+      listing manifests...
+    </div>
+    <div v-if="getManifestErrors.list" class="build-error">
+      <p>An error has occured during the retrieving of manifests</p>
```
```suggestion
      <p>An error occurred while retrieving the list of manifests</p>
```

Arignir commented on front/src/views/manifest/update.vue:
```diff
@@ -0,0 +1,333 @@
+        <h1>Update manifest</h1>
+        <!-- Error handling -->
+        <div v-if="getManifestLoadings.update" class="loading">
+          updating the manifest...
```
```suggestion
          Updating the manifest...
```

Arignir commented on front/src/views/manifest/update.vue:
```diff
@@ -0,0 +1,333 @@
+          updating the manifest...
+        </div>
+        <div v-if="error || getManifestErrors.update" class="build-error">
+          <p>An error has occured during the update of the manifest</p>
```
```suggestion
          <p>An error occurred while updating the manifest</p>
```

Arignir commented on front/src/views/manifest/update.vue:
```diff
@@ -0,0 +1,333 @@
+        </div>
+
+        <div v-if="getManifestLoadings.get" class="loading">
+          loading
```
```suggestion
          Loading...
```

Arignir commented on front/src/views/manifest/update.vue:
```diff
@@ -0,0 +1,333 @@
+          loading
+        </div>
+        <div v-else-if="getManifestErrors.get" class="build-error">
+          <p>An error has occured, please refresh or try again later</p>
```
```suggestion
          <p>An error occurred, please try again later</p>
```

Arignir reviewed (changes requested):
There's a lot of things that can be improved in this PR (there's a lot of code duplication for example), but I assume most of this code will be revamped soon with the new front end so I assume it's fine

BourgoisMickael committed:
[Front] Fix for PR review

Rewrote messages (error/loading) and other texts
Removed blank lines
Marked some todos for next PR
Use enumerations to replace magic values

Co-Authored-By: Arignir <Arignir@users.noreply.github.com>

BourgoisMickael committed:
[README] Added root README with link for subfolders
